File: 2750-ways-to-split-array-into-good-subarrays/2750-ways-to-split-array-into-good-subarrays.py
class Solution:
    def numberOfGoodSubarraySplits(self, nums: List[int]) -> int:
        total = sum(nums)
        if total == 0:
            return 0
        if total == 1:
            return 1
        
        res = 1
        s = False
        count = 0
        for i in range(len(nums)):
            if nums[i] == 1 and s == False:
                s = True
            elif nums[i] == 0 and s == True:
                count += 1
            elif nums[i] == 1 and s == True:
            
                res *= (count+1)
                count = 0
        return res % (10**9+7)

File: 2750-ways-to-split-array-into-good-subarrays/2750-ways-to-split-array-into-good-subarrays.cs
public class Solution {
    public int NumberOfGoodSubarraySplits(int[] nums) {
        var total = nums.Sum();
        if (total == 0)
            return 0;
        long res = 1;
        int count = 0;
        bool s = false;
        long mod = (long)(Math.Pow(10, 9)+7);
        
        foreach (int n in nums) {
            if (n == 1 && s == false) {
                s = true;
            } else if (n == 0 && s == true) {
                count += 1;
            } else if (n == 1 && s == true) {
                res = (res*(count + 1)) % mod;
                count = 0;
            }
        }
        return (int)res;
        
    }
}